### PR TITLE
community: Add async methods to CassandraVectorStore

### PR DESCRIPTION
**Pull request by cbornet**

### CODE REVIEW

File: libs/community/pyproject.toml [hunk 1]
```diff
@@ -283,13 +283,13 @@ exclude = [
...
-  "I",  # isort
-  "T201", # print
-]
+#[tool.ruff.lint]
+#select = [
+#  "E",  # pycodestyle
+#  "F",  # pyflakes
+#  "I",  # isort
+#  "T201", # print
+#]
```
Comment (baskaryan) [inline libs/community/pyproject.toml:292]: uncomment

File: pyproject.toml [hunk 2]
```diff
@@ -72,16 +72,16 @@ extend-exclude = [
...
+#    "E402",  # allow imports to appear anywhere in docs
+#    "F401",  # allow "imported but unused" example code
+#    "F811",  # allow re-importing the same module, so that cells can stay independent
+#    "F841",  # allow assignments to variables that are never read -- it's example code
+#]
+#
+## These files were failing the listed rules at the time ruff was adopted for notebooks.
+## Don't require them to change at once, though we should look into them eventually.
+#"cookbook/gymnasium_agent_simulation.ipynb" = ["F821"]
+#"docs/docs/integrations/document_loaders/tensorflow_datasets.ipynb" = ["F821"]
```
[the remark this answers is not in the document]
Comment (cbornet) [reply pyproject.toml:86]: oops

[the remark this answers is not in the document]
Comment (cbornet) [reply pyproject.toml]: fixed.

File: libs/community/pyproject.toml (hunk 1 above)
Comment (cbornet) [reply libs/community/pyproject.toml]: fixed.

**Review by eyurtsev — APPROVED** [review]

**Pull request merged**